ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -247,19 +248,23 @@ public SecurityIdentity getSecurityIdentity() {
     private void updateAffinities(InvocationRequest invocationRequest, Map<String, Object> attachments, EJBLocator<?> ejbLocator, ComponentView componentView) {
         Affinity legacyAffinity = null;
         Affinity weakAffinity = null;
+        Affinity strongAffinity = null;
         Affinity clusterAffinity = getClusterAffinity();
 
         if (ejbLocator.isStateful() && componentView.getComponent() instanceof StatefulSessionComponent) {
             final StatefulSessionComponent statefulSessionComponent = (StatefulSessionComponent) componentView.getComponent();
+            strongAffinity = getStrongAffinity(statefulSessionComponent);
             weakAffinity = legacyAffinity = getWeakAffinity(statefulSessionComponent, ejbLocator.asStateful());
         } else if (componentView.getComponent() instanceof StatelessSessionComponent) {
-            // V3 and less used cluster affinity as a weak affinity for SLSBs
+            // V3 and less used cluster affinity as a weak affinity for SLSBs (used to update weak affinity in the invocation context on invocation return)
             legacyAffinity = clusterAffinity;
+            // for SLSB, if they touch a cluster, assign them ClusterAffinity (to retain compatability with what was going on before)
+            strongAffinity = clusterAffinity;
```

Comment:
strongAffinity = getClusterAffinity();

Reply:
Ideally, SLSBs have no strong affinity.  Just a weak affinity to the cluster on which the bean was created.  But I understand if this is needed for compatibility reasons.